plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseElasticsearchConnectorTest.java

```diff
@@ -1860,6 +1861,69 @@ public void testMissingIndex()
+                        "FROM TABLE(elasticsearch.system.query(" +
+                        "schema => 'tpch', " +
+                        "index => 'nation', " +
+                        "query => '{\"query\": {\"match\": {\"name\": \"ALGERIA\"}}}')) t(result)",
+                "VALUES '{\"nationkey\":0,\"name\":\"ALGERIA\",\"regionkey\":0,\"comment\":\" haggle. carefully final deposits detect slyly agai\"}'");
+
+        // parameters
+        Session session = Session.builder(getSession())
+                .addPreparedStatement(
+                        "my_query",
+                        "SELECT json_query(result, 'lax $[0][0].hits.hits._source') FROM TABLE(elasticsearch.system.query(schema => ?, index => ?, query => ?))")
+                .build();
+        assertQuery(
+                session,
+                "EXECUTE my_query USING 'tpch', 'nation', '{\"query\": {\"match\": {\"name\": \"ALGERIA\"}}}'",
+                "VALUES '{\"nationkey\":0,\"name\":\"ALGERIA\",\"regionkey\":0,\"comment\":\" haggle. carefully final deposits detect slyly agai\"}'");
+
+        // select multiple records by range. Use array wrapper to wrap multiple results
+        assertQueryMatchesAny("SELECT json_query(result, 'lax $[0][0].hits.hits._source.name' WITH ARRAY WRAPPER) " +
+                        "FROM TABLE(elasticsearch.system.query(" +
+                        "schema => 'tpch', " +
+                        "index => 'nation', " +
+                        "query => '{\"query\": {\"range\": {\"nationkey\": {\"gte\": 0,\"lte\": 1}}}}')) t(result)",
+                ImmutableList.of(
+                        "VALUES '[\"ALGERIA\",\"ARGENTINA\"]'",
+                        "VALUES '[\"ARGENTINA\",\"ALGERIA\"]'"));
+
+        // no matches
+        assertQuery("SELECT json_query(result, 'lax $[0][0].hits.hits') " +
+                        "FROM TABLE(elasticsearch.system.query(" +
+                        "schema => 'tpch', " +
+                        "index => 'nation', " +
+                        "query => '{\"query\": {\"match\": {\"name\": \"UTOPIA\"}}}')) t(result)",
+                "VALUES '[]'");
+
+        // syntax error
+        assertThatThrownBy(() -> query("SELECT * " +
+                "FROM TABLE(elasticsearch.system.query(" +
+                "schema => 'tpch', " +
+                "index => 'nation', " +
+                "query => 'wrong syntax')) t(result)"))
+                .hasMessageContaining("json_parse_exception");
+    }
+
+    private void assertQueryMatchesAny(String actual, List<String> expectedOptions)
+    {
+        for (String expected : expectedOptions) {
+            try {
+                assertQuery(actual, expected);
+                return;
+            }
+            catch (AssertionError e) {
+                // ignore, try other options
+            }
+        }
+        throw new AssertionError("Neither of expected options matched the actual result");
+    }
```

Comment:
I don't think this is such a good idea. It masks the underlying assertion, so it's hard to tell what was different and it what way. It also doesn't scale when there are more than a couple of alternatives.

Instead, modify the query to normalize the results in some way. For instance, parse the resulting json, extract the values and sort them to get a consistent result.